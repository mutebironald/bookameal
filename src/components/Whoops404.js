import React from "react";

export default function Whoops404({ location }) {
  return (
    <div className="whoops-404">
      <h1>
        Resource not found at'{location.pathname}'
      </h1>
    </div>
  );
}
