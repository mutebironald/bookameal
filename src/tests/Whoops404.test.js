import React from 'react';
import Whoops404 from '../components/Whoops404';


describe("Not found component", () => {
    let wrapper;
    beforeEach(() => {
        const location = {
            pathname: ""
        }
        wrapper = shallow(<Whoops404 location={location} />);
        
    });
    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
