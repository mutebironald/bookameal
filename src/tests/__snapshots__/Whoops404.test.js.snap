// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not found component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Whoops404
    location={
        Object {
            "pathname": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Resource not found at'
        
        '
</h1>,
      "className": "whoops-404",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Resource not found at'",
          "",
          "'",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Resource not found at'",
        "",
        "'",
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Resource not found at'
          
          '
</h1>,
        "className": "whoops-404",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Resource not found at'",
            "",
            "'",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Resource not found at'",
          "",
          "'",
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
